src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1191 @@
+    REAL(shr_kind_r8)               :: outTmp_3D(State_Grid%NY, State_Grid%NZ)
+    REAL(shr_kind_r8)               :: outTmp_2D(State_Grid%NY)
+
+    ! ================================================================
+    ! CopyGCStates2Exports begins here
+    ! ================================================================
+    __Iam__('CopyGCStates2Exports (cesmgc_history_mod.F90)')
+    RC = GC_SUCCESS
+
+    ! Loop over the History Exports list
+    current => HistoryConfig%HistoryExportsList%head
+    DO WHILE ( ASSOCIATED( current ) )
+       ! Skip if not active
+       if(.not. hist_fld_active(trim(current%name))) then
+          current => current%next
+          cycle
+       endif
+
+       ! if (am_I_Root) THEN
+       !    print *, '  Copying ' // TRIM(current%name)
+       ! endif
+       IF ( current%rank == 2 ) THEN
+          IF ( ASSOCIATED( current%GCStateData2d ) ) THEN
+             outTmp_2D(1:State_Grid%NY) = current%GCStateData2d(1,1:State_Grid%NY)
+          ELSE IF ( ASSOCIATED( current%GCStateData2d_4 ) ) THEN
+             outTmp_2D(1:State_Grid%NY) = current%GCStateData2d_4(1,1:State_Grid%NY)
+          ELSE IF ( ASSOCIATED( current%GCStateData2d_8 ) ) THEN
+             outTmp_2D(1:State_Grid%NY) = current%GCStateData2d_8(1,1:State_Grid%NY)
+          ELSE IF ( ASSOCIATED( current%GCStateData2d_I ) ) THEN
+             ! Convert integer to float (integers not allowed in MAPL exports)
+             outTmp_2D(1:State_Grid%NY) = FLOAT(current%GCStateData2d_I(1,1:State_Grid%NY))
+          ELSE
+             RC = GC_FAILURE
+             ErrMsg = "No GC 2D pointer found for " // TRIM(current%name)
+             EXIT
+          ENDIF
+
+          ! Now call outfld to output for this chunk
+          call outfld(trim(current%name),                              &
+                      outTmp_2D,                                       &   ! Chunk width always 1
+                      State_Grid%NY,                                   &
+                      LCHNK                                           )
+       ELSEIF ( current%rank == 3 ) THEN
+          IF ( ASSOCIATED( current%GCStateData3d ) ) THEN
+             outTmp_3D(1:State_Grid%NY, :) = current%GCStateData3d(1,1:State_Grid%NY,:)
+          ELSE IF ( ASSOCIATED( current%GCStateData3d_4 ) ) THEN
+             outTmp_3D(1:State_Grid%NY, :) = current%GCStateData3d_4(1,1:State_Grid%NY,:)
+          ELSE IF ( ASSOCIATED( current%GCStateData3d_8 ) ) THEN
+             outTmp_3D(1:State_Grid%NY, :) = current%GCStateData3d_8(1,1:State_Grid%NY,:)
+          ELSE IF ( ASSOCIATED( current%GCStateData3d_I ) ) THEN
+             outTmp_3D(1:State_Grid%NY, :) = FLOAT(current%GCStateData3d_I(1,1:State_Grid%NY,:))
+          ELSE
+             RC = GC_FAILURE
+             ErrMsg = "No GC 3D pointer found for " // TRIM(current%name)
+             EXIT
+          ENDIF
+#if defined( MODEL_GEOS ) || defined( MODEL_CESM )
```

Comment:
Why is this cpp block here? This will only ever be built in CAM.

Reply:
Makes diff easier. I've changed so it only matches `MODEL_CESM` now.